lib/platform.c

```diff
@@ -543,20 +546,24 @@ int STDCALL sf_delete_directory_if_exists(const char * directoryName)
 void STDCALL sf_get_tmp_dir(char * tmpDir)
 {
 #ifdef _WIN32
+#pragma comment(lib, "rpcrt4.lib")
+  UUID uuid;
+  UuidCreate(&uuid);
+  unsigned char* uuid_cstr = NULL;
+  UuidToStringA(&uuid, (RPC_CSTR*)&uuid_cstr);
   GetTempPath(100, tmpDir);
+  sprintf(tmpDir+strlen(tmpDir),"%s\\",uuid_cstr); 
+  RpcStringFreeA((RPC_CSTR*)(&uuid_cstr));
 #else
   const char * tmpEnv = getenv("TMP") ? getenv("TMP") : getenv("TEMP");
-
+  uuid_t uuid;
+  char uuid_cstr[37]; // 36 byte uuid plus null.
+  uuid_generate(uuid);
+  uuid_unparse(uuid, uuid_cstr);
   if (!tmpEnv)
-  {
-    strncpy(tmpDir, "/tmp/", sizeof("/tmp/"));
-  }
+    sprintf(tmpDir,"/tmp/%s/",uuid_cstr);
   else
-  {
-    strncpy(tmpDir, tmpEnv, strlen(tmpEnv));
-    size_t oldLen = strlen(tmpDir);
-    tmpDir[oldLen] = PATH_SEP;
-    tmpDir[oldLen+1] = '\0';
-  }
+    sprintf(tmpDir, "%s/%s/",tmpEnv,uuid_cstr);
 #endif
-}
\ No newline at end of file
+  sf_create_directory_if_not_exists(tmpDir);
```

Comment:
If the directory exists, it is very likely the tmp directory is already used by other PUT commands?

Reply:
This is going to be a unique temp directory for every put command. Every time the customer tries to put file we create a unique temp directory. which looks like
/tmp/<uuid string>/